plugins/v1/repository_s3/plugin.py

```diff
@@ -0,0 +1,113 @@
+# Licensed to Elasticsearch B.V. under one or more contributor
+# license agreements. See the NOTICE file distributed with
+# this work for additional information regarding copyright
+# ownership. Elasticsearch B.V. licenses this file to you under
+# the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#	http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import logging
+import os
+import shlex
+import subprocess
+
+from esrally.utils import process
+from esrally import exceptions
+
+LOGGER_NAME = "rally.provisioner.repository_s3"
+
+
+def resolve_binary(install_root, binary_name):
+    return os.path.join(install_root, "bin", binary_name)
+
+
+def resolve_keystore_config(install_root):
+    return os.path.join(install_root, "config", "elasticsearch.keystore")
+
+
+def create_keystore(install_root, keystore_binary, env):
+    logger = logging.getLogger(LOGGER_NAME)
+
+    keystore_create_command = "{keystore} -s create".format(keystore=keystore_binary)
+
+    return_code = process.run_subprocess_with_logging(
+        keystore_create_command,
+        env=env
+    )
+
+    if return_code != 0:
+        logger.error("%s has exited with code [%d]", keystore_create_command, return_code)
+        raise exceptions.SystemSetupError(
+            "Could not initialize a keystore. Please see the log for details.")
+
+
+def add_property_to_keystore(keystore_binary, client_name, property_name, property_value, env):
+    logger = logging.getLogger(LOGGER_NAME)
+
+    p1 = subprocess.Popen(["echo", property_value], stdout=subprocess.PIPE)
+
+    keystore_command = "{keystore} --silent add --stdin s3.client.{client_name}.{key}".format(
+        keystore=keystore_binary,
+        client_name=client_name,
+        key=property_name,
+        value=property_value)
```

Comment:
The named parameter `value` is unused and can be removed.

Reply:
Great catch, thanks, fixed in db748c6098d76529876ef5deac3eb35c73c9a4a0